### PR TITLE
Manage {system,user}.conf values

### PR DESCRIPTION
**Pull request by jay7x**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -219,6 +219,22 @@
 #
 # @param udev_purge_rules
 #   Toggle if unmanaged files in /etc/udev/rules.d should be purged if manage_udevd is enabled
+#
+# @param manage_system_conf
+#   Should system service manager configurations be managed
+#
+# @param system_settings
+#   Config Hash that is used to configure settings in system.conf
+#   NOTE: It's currently impossible to have multiple entries of the same key in
```
Comment (kenyon) [inline manifests/init.pp:228]: https://github.com/puppetlabs/puppetlabs-inifile/issues/41

Comment (jay7x) [reply manifests/init.pp:228]: Yep, I found it.. still not implemented 🤷🏻‍♂️

File: types/servicemanagersettings.pp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    Optional['DefaultLimitMEMLOCK'] => Variant[Pattern['^(infinity|((\d+(K|M|G|T|P|E)?(:\d+(K|M|G|T|P|E)?)?)))$'], Systemd::SettingEnsure],
+    Optional['DefaultLimitLOCKS'] => Variant[Integer[1], Systemd::SettingEnsure],
+    Optional['DefaultLimitSIGPENDING'] => Variant[Integer[1], Systemd::SettingEnsure],
+    Optional['DefaultLimitMSGQUEUE'] => Variant[Pattern['^(infinity|((\d+(K|M|G|T|P|E)?(:\d+(K|M|G|T|P|E)?)?)))$'], Systemd::SettingEnsure],
+    Optional['DefaultLimitNICE'] => Variant[Integer[0,40], Pattern['^(-\+([0-1]?[0-9]|20))|([0-3]?[0-9]|40)$'], Systemd::SettingEnsure],
+    Optional['DefaultLimitRTPRIO'] => Variant[Integer[0], Systemd::SettingEnsure],
+    Optional['DefaultLimitRTTIME'] => Variant[Pattern['^\d+(ms|s|m|h|d|w|M|y)?(:\d+(ms|s|m|h|d|w|M|y)?)?$'], Systemd::SettingEnsure],
+    Optional['DefaultOOMPolicy'] => Variant[Enum['continue', 'stop','kill'], Systemd::SettingEnsure],
+    Optional['DefaultSmackProcessLabel'] => Variant[String, Systemd::SettingEnsure],
+    Optional['ReloadLimitIntervalSec'] => Variant[Enum['infinity'], Systemd::Timespan, Systemd::SettingEnsure],
```
Comment (traylenator) [inline types/servicemanagersettings.pp:70]: Can you elaborate on your thinking here for the value `=> present` or  `=> absent` here ?

Comment (jay7x) [reply types/servicemanagersettings.pp:70]: Copied it from other places (see `journald.pp` and `journaldsettings.pp` e.g.). As I understand this is the way to remove a record from a file:

```yaml
systemd::system_settings:
  ReloadLimitIntervalSec:
    ensure: absent
```

Cannot say I like it though.. especially, how it's implemented (with overriding things via `Ini_setting[$title]`)

Comment (traylenator) [reply types/servicemanagersettings.pp:70]: > Copied it from other places (see `journald.pp` and `journaldsettings.pp` e.g.). As I understand this is the way to remove a record from a file:

I had never noticed that before on journaldsettings.

File: types/timespan.pp [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# @summary Defines a timespan type
+# @see https://www.freedesktop.org/software/systemd/man/latest/systemd.time.html
+type Systemd::Timespan = Variant[
+  Integer[0],
+  Pattern[/^([0-9]+ *(usec|us|msec|ms|second[s]?|sec|s|minute[s]?|min|m|hour[s]?|hr|h|day[s]?|d|week[s]?|w|month[s]?|M|year[s]?|y)? *)+$/]
```
Comment (kenyon) [inline types/timespan.pp:5]: The character classes for the plurals aren't needed:

```suggestion
  Pattern[/^([0-9]+ *(usec|us|msec|ms|seconds?|sec|s|minutes?|min|m|hours?|hr|h|days?|d|weeks?|w|months?|M|years?|y)? *)+$/]
```

Comment (jay7x) [reply types/timespan.pp]: TY! Committed your suggestion :)

**Review by kenyon — APPROVED** [review]

File: manifests/init.pp [hunk 4]
```diff
@@ -188,10 +188,13 @@
...
 #
 # @param manage_accounting
-#   when enabled, the different accounting options (network traffic, IO, CPU util...) are enabled for units
+#   When enabled, the different accounting options (network traffic, IO,
+#   CPU util...) are enabled for units.
 #
 # @param accounting
-#   Hash of the different accounting options. This highly depends on the used systemd version. The module provides sane defaults per operating system using Hiera.
+#   Hash of the different accounting options. This highly depends on the used
+#   systemd version. The module provides sane defaults per operating system
```
Comment (bastelfreak) [inline manifests/init.pp:196]: I think if we drop all EoL OSes, the remaining OSes all use the same accounting options.

**Review by bastelfreak — APPROVED** [review]

File: manifests/init.pp (hunk 4 above)
Comment (jay7x) [reply manifests/init.pp:196]: That's what I hope for also.. but it should be done in another PR I guess

**Pull request merged**